packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -59,23 +60,28 @@ export function getProxyForField(field: FlexTreeField): TreeNode | TreeValue | u
 		case FieldKinds.required: {
 			const asValue = field as FlexTreeTypedField<FlexFieldSchema<typeof FieldKinds.required>>;
 
-			// TODO: Ideally, we would return leaves without first boxing them.  However, this is not
-			//       as simple as calling '.content' since this skips the node and returns the FieldNode's
-			//       inner field.
-			return getOrCreateNodeProxy(asValue.boxedContent);
+			const maybeUnboxedContent = asValue.content;
+			if (maybeUnboxedContent === undefined) {
+				return undefined;
+			}
+			if (!isFlexTreeNode(maybeUnboxedContent)) {
+				return maybeUnboxedContent;
+			}
+
+			return getOrCreateNodeProxy(maybeUnboxedContent);
```

Comment:
Its logically simpler to skip the negation and swap the cases:
```suggestion
			if (isFlexTreeNode(maybeUnboxedContent)) {
				return getOrCreateNodeProxy(maybeUnboxedContent);
			}

			return maybeUnboxedContent;
```


Also since thuis case is the same as the required one they can be combined with a case fallthrouhg, which saves some code:

```
case FieldKinds.required:
case FieldKinds.optional: {
			const asValue = field as FlexTreeTypedField<FlexFieldSchema<typeof FieldKinds.required | typeof FieldKinds.required>>;
```

Reply:
I did consider the swapping, but it felt a bit more natural conceptually to try to handle the (arguably) simpler case of a plain value first, and then fallback to treating it like a node. No super strong feelings though. And now that the undefined case is gone from both, the swap doesn't feel as unnatural, so I'll apply it and consolidate optional and required.

Reply:
Ended up not consolidating the cases entirely but extracting the logic to a function. In each case the field is cast differently (e.g. `field as FlexTreeTypedField<FlexFieldSchema<typeof FieldKinds.required>>` vs `...<typeof FieldKinds.optional>`). Which can be used equivalently here, but I didn't want to cast as one or the other and leave a potential bug later, when someone did something that wasn't allowed for one or the other field kinds.